l10n_it_fatturapa_out/tests/test_fatturapa_out_noteline.py

```diff
@@ -0,0 +1,196 @@
+            line_form.name = "just a note"
+            line_form.account_id = self.env["account.account"]
+        invoice = move_form.save()
+
+        res = self.run_wizard(invoice.id)
+        self.assertTrue(res)
+
+        attachment = self.attach_model.browse(res["res_id"])
+        xml_content = base64.decodebytes(attachment.datas)
+
+        parser = etree.XMLParser(remove_blank_text=True)
+        xml = etree.fromstring(xml_content, parser)
+        aliquote = xml.findall(".//DatiRiepilogo/AliquotaIVA")
+        self.assertEqual(len(aliquote), 1)
+        self.assertEqual(aliquote[0].text, "22.00")
+
+    def test_3_note_taxes(self):
+        """test case:
+        - company default 22%
+        - one product 22%
+        - one product 10%
+        - one note
+
+        expected result:
+        - one DatiRiepilogo with AliquotaIVA == 10%
+        - one DatiRiepilogo with AliquotaIVA == 22%
+        (order irrelevant)
+        """
+        self.env.company.account_sale_tax_id = self.tax_22
+
+        move_form = Form(
+            self.env["account.move"].with_context(default_move_type="out_invoice")
+        )
+        move_form.name = "INV/2016/0013"
+        move_form.invoice_date = fields.Date.from_string("2016-01-07")
+        move_form.partner_id = self.res_partner_fatturapa_0
+        with move_form.invoice_line_ids.new() as line_form:
+            line_form.product_id = self.product_product_10
+            line_form.tax_ids.remove(index=0)
+            line_form.tax_ids.add(self.tax_22)
+        with move_form.invoice_line_ids.new() as line_form:
+            line_form.product_id = self.product_product_10
+            line_form.tax_ids.remove(index=0)
+            line_form.tax_ids.add(self.tax_10)
+        with move_form.invoice_line_ids.new() as line_form:
+            line_form.display_type = "line_note"
+            line_form.name = "just a note"
+            line_form.account_id = self.env["account.account"]
+        invoice = move_form.save()
+
+        res = self.run_wizard(invoice.id)
+        self.assertTrue(res)
+
+        attachment = self.attach_model.browse(res["res_id"])
+        xml_content = base64.decodebytes(attachment.datas)
+
+        parser = etree.XMLParser(remove_blank_text=True)
+        xml = etree.fromstring(xml_content, parser)
+        aliquote = xml.findall(".//DatiRiepilogo/AliquotaIVA")
+        self.assertEqual(len(aliquote), 2)
+        if aliquote[0].text == "22.00":
+            self.assertEqual(aliquote[1].text, "10.00")
+        else:
+            self.assertEqual(aliquote[0].text, "10.00")
+            self.assertEqual(aliquote[1].text, "22.00")
+
+    def test_4_note_taxes(self):
+        """test case:
+        - company default 22%
+        - one product 22%, price 10
+        - one note
+        save & edit
+        - produce price to 0
+        save & export
+
+        expected result:
+        - one DatiRiepilogo with AliquotaIVA == 22%
+        """
```

Comment:
Come mai questo caso dovrebbe avere un risultato diverso da `test_2_note_taxes`?
Cosa sospetti possa andare storto modificando la riga prodotto?

Reply:
In teoria nulla, ma è l'ultima casistica di fallimento riportata (https://github.com/OCA/l10n-italy/pull/2335#issuecomment-921919080) e ho voluto testare quel particolare flusso così come riportato.